modin/pandas/base.py

```diff
@@ -3291,6 +3265,9 @@ def __len__(self):
     def __lt__(self, right):
         return self.lt(right)
 
+    def __matmul__(self, other):
```

Comment:
If we want this to be a part of this PR, we need to add a test.

Reply:
Added tests both for `DataFrame` and `Series`